packages/ui/src/components/WatchedAccounts.tsx

```diff
@@ -36,7 +36,7 @@ const WatchedAccounts = ({ className }: Props) => {
         })}
       </ListStyled>
       <TitleStyled>Watch new account...</TitleStyled>
-      <AccountSelectionWrapperStyled>
+      <AccountSelectionWrapperStyled data-cy="watched-account-wrapper">
```

Comment:
I would suggest using Testing library with Cypress, as it allows you to write the selectors that aren't id specific, rather the role. During the test run it assigns the role and you can find an element without artificially creating "data-cy" parameter which is used only by the test. Also, there is a plugin that suggests to you what is the proper selector to choose for certain elements.  (https://chrome.google.com/webstore/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano)

Reply:
I'm not sure I get it. `data-cy` is actually best practice https://docs.cypress.io/guides/references/best-practices#Selecting-Elements We do want to be very specific, and not rely on elements or classes, or any other id that could be duplicated or used for anything else than testing.

Or did I miss you point?

Reply:
@Tbaut 
It's exactly the problem that the library will solve, you don't need to add anything manually even the data-cy ids. It will add the "role" attribute on the run based on the elements (f.e it will assign a role button to the buttons that you can find with the following selector: cy.findByRole('button', 'connect')). Testing library it's actually a gold standard and can be used almost with any frontend framework, It worth the time to learn it once and use it for any project. Here is how to set it up with Cypress https://testing-library.com/docs/cypress-testing-library/intro

The main benefit is that it forces you to write the tests in the way that you interact with visual data, elements, and text and test the app with the way the user will see it. With data- props you can assign it even to the invisible element and still the test will pass as it's present in DOM, but visually it might not appear on the page. More information here:
https://testing-library.com/docs/#the-problem 


Regarding the playground plugin:
It's just a helper tool if you struggle to find a meaningful selector for the element, You can try it here (https://testing-playground.com/), hover over the element and it will show you in the green box how is a good query could look like.

Reply:
Thanks a lot for the detailed, explanation. I think this can help @asnaith a lot, so that you don't have to actually add some `data-cy` all over. @asnaith can you just tell me if you're good with using this, and I'll add it in this PR, and rewrite that first test using it.